SDD-766 Correct logic in filling empty slots when decription ids...

AAAlinaaa opened the pull request:
... are included in refset dsv export

AAAlinaaa committed:
SDD-766 Correct logic in filling empty slots when decription ids...

... are included in refset dsv export

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/refset/SnomedSimpleTypeRefSetDSVExporter.java:
```diff
@@ -320,7 +320,12 @@ public boolean apply(ISnomedDescription input) {
 								
-								stringBuffer.append(joinResultsWithDelimiters(results, exportItemMaxOccurences.get(descriptionTypeId), delimiter, descriptionIdExpected));
+								int maxOccurences = exportItemMaxOccurences.get(descriptionTypeId);
+								if(descriptionIdExpected) {
```
Whitespace!

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/refset/SnomedSimpleTypeRefSetDSVExporter.java:
```diff
@@ -320,7 +320,12 @@ public boolean apply(ISnomedDescription input) {
-								stringBuffer.append(joinResultsWithDelimiters(results, exportItemMaxOccurences.get(descriptionTypeId), delimiter, descriptionIdExpected));
+								int maxOccurences = exportItemMaxOccurences.get(descriptionTypeId);
+								if(descriptionIdExpected) {
+									maxOccurences = maxOccurences*2;
```
This should be moved to joinResultsWithDelimiters

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/refset/SnomedSimpleTypeRefSetDSVExporter.java:
```diff
@@ -1029,9 +1034,6 @@ private String joinResultsWithDelimiters(Collection<String> results, int max, St
 			sb.append(delimiter);
-			if (descriptionAndDescriptionIdExpected) {
-				sb.append(delimiter);
-			}
```
This removal will fix description export but will break relationship export

nagyo reviewed (changes requested):
Let me know before you start fixing this, I have a few comments for the implementation.

AAAlinaaa committed:
SDD-766 Refactor delimiter distribution in dsv export

1. join ids and labels to be treated as one unit in
joinResultsWithDelimeters
2. recalculate number of delimeters needed when description or
relationship ids are used
3. in case there is no result to display, the delimeter at the beginning
of every export item will take the place of the first delimeter

nagyo commented on snomed/com.b2international.snowowl.snomed.exporter.server/src/com/b2international/snowowl/snomed/exporter/server/refset/SnomedSimpleTypeRefSetDSVExporter.java:
```diff
@@ -316,16 +315,11 @@ public boolean apply(ISnomedDescription input) {
 								}).transformAndConcat(new Function<ISnomedDescription, List<String>>() {
```
You can replace this transformAndConcat with a simple transform and get rid of the singleton lists, since now we are using single items in any case.

nagyo reviewed (changes requested):
Fix that one last thing and it's ready to be merged!

[![LGTM](http://lgtm.in/p/6vYSApsUp)](http://lgtm.in/i/6vYSApsUp)

AAAlinaaa committed:
SDD-766: Replace transformAndConcat with transform while collecting...

... descriptions

nagyo reviewed (approved):
[![LGTM](http://lgtm.in/p/go0DwoFiq)](http://lgtm.in/i/go0DwoFiq)
